go/vt/vtgate/executor.go

```diff
@@ -207,9 +207,10 @@ func (e *Executor) execute(ctx context.Context, safeSession *SafeSession, sql st
 		return 0, nil, err
 	}
 
-	if safeSession.InTransaction() && destTabletType != topodatapb.TabletType_MASTER {
-		return 0, nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "transactions are supported only for master tablet types, current type: %v", destTabletType)
-	}
+	// TODO(deepthi): we should check for the right condition that allows transactions
+	//if safeSession.InTransaction() && destTabletType != topodatapb.TabletType_MASTER {
```

Comment:
Is it correct to just remove this check? or should we be checking that if InTransaction is true we have a transactionID and tabletAlias to use in the transaction?

Reply:
with @systay's change to `start transaction read only` it should be safe to check these conditions only for legacy gateway.